docs/weave/tutorial/11-cron.md

```diff
@@ -0,0 +1,325 @@
+Instead, the signature check step is bypassed and desired conditions are set in the context when
+processing given scheduled message.
+
+Absence of signature could be security hole if a developer implements the CRON feature poorly.
+Happily only code can schedule a CRON task and decide on conditions, so this should never be an
+issue.
+
+### Ticker
+
+[Ticker interface](https://github.com/iov-one/weave/blob/master/cron.go#L11-L27) defines the API
+of the executor of the scheduled jobs.
+
+```go
+// Ticker is an interface used to call background tasks scheduled for
+// execution.
+type Ticker interface {
+    // Tick is a method called at the beginning of the block. It should be
+    // used to execute any scheduled tasks.
+    //
+    // Because beginning of the block does not allow for an error response
+    // this method does not return one as well. It is the implementation
+    // responsibility to handle all error situations.
+    // In case of an error that is an instance specific (ie database
+    // issues) it might be neccessary for the method to terminate (ie
+    // panic). An instance specific issue means that all other nodes most
+    // likely succeeded processing the task and have different state than
+    // this instance. This means that this node is out of sync with the
+    // rest of the network and cannot continue operating as its state is
+    // invalid.
+    Tick(ctx Context, store CacheableKVStore) TickResult
+}
+```
+
+## Tutorial
+
+### Message
+
+In [CreateArticleMsg](https://github.com/iov-one/blog-tutorial/blob/master/x/blog/codec.proto#L92-L104)
+we defined a field named as `delete_at` that defines the deletion time of the article that will
+be initated by CRON. During the creation of CreateArticleHandler scheduler is defined as a part
+of the struct:
+
+### Scheduler Handler
+
+```go
+type CreateArticleHandler struct {
+    auth      x.Authenticator
+    ab        *ArticleBucket
+    bb        *BlogBucket
+    dtb       *DeleteArticleTaskBucket
+    scheduler weave.Scheduler
+}
+```
+
+_weave.Scheduler_ will be initiated and passed to the handler on the application layer.
+
+[The code](https://github.com/iov-one/blog-tutorial/blob/master/x/blog/handler.go#L358-L363) that
+schedules a job to be processed in future is below:
+
+```go
+// schedule delete task
+	if msg.DeleteAt != 0 {
+		deleteArticleMsg := &DeleteArticleMsg{
+			Metadata:  &weave.Metadata{Schema: 1},
+			ArticleID: article.ID,
+		}
+
+		var taskID []byte
+		taskID, err = h.scheduler.Schedule(store, article.DeleteAt.Time(), nil, deleteArticleMsg)
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot schedule deletion task")
+		}
+
+		// save delete article task so it could be cancelled later
+		deleteArticleTask := &DeleteArticleTask{
+			Metadata:  deleteArticleMsg.Metadata,
+			ID:        taskID,
+			ArticleID: article.ID,
+			TaskOwner: x.MainSigner(ctx, h.auth).Address(),
+		}
```

Comment:
Here you create an entity for the deletion taks. This is not wrong, although might be not necessary.

In `x/gov` package, proposal entity contains information of a related cron task. Including reference in the entity this task is about might make the relation and code structure easier.

https://github.com/iov-one/weave/blob/master/x/gov/codec.proto#L147

Reply:
I am aware of that implementation, but a separate entity seemed like a better solution that makes readers of the docs understand task concept.